data_model/src/lib.rs

```diff
@@ -336,24 +331,25 @@ pub enum IdBox {
 )]
 pub enum RegistrableBox {
```

Comment:
I'm thinking I would rename to `RegisteredWithBox` to be consistent or rename `RegisterWith` trait to `Registrable`

Reply:
`Registrable::With`? Shorter and we almost always use the `<Self as Registrable>::With`.

Reply:
sounds good